### DISCUSSION TITLE
500 Internal Server Error, Nextjs app with Auth0

### DISCUSSION THREAD

**Discussion by bryanyanay** [vercel/community#766] [Help]
Hello! I recently deployed my nextjs app that uses auth0 for authentication to vercel. The app works perfectly on localhost, but not on vercel.
 - [link to repo](https://github.com/bryanyanay/power-transfer-app/tree/main)
 - [link to app](https://power-transfer-app.vercel.app/)

Right now for the production vercel deployment, logging in and out is working, however, when I try to access one of the pages (transfer, history, or vehicles), I get a page with the following: 
![image](https://user-images.githubusercontent.com/74737261/182889779-14363a4d-cf02-4e7c-a7e5-607d15d67a9b.png)

And I'm not sure why this is happening. Is someone here able to help me figure this out? If it is relevant, here's some extra info:
 - here is the error in vercel's function logs when i try to access the history page:
![image](https://user-images.githubusercontent.com/74737261/182894928-1559913e-8c1a-4ac8-8312-8d94870867de.png)
 - I'm using yugabyteDB as the database, and prisma as the ORM.
 - here are the environment variables I've set in vercel:
![image](https://user-images.githubusercontent.com/74737261/182890022-c77a7329-7d5f-4b74-8eef-767c684f1108.png)
 - and here are my settings on the auth0 app:
![image](https://user-images.githubusercontent.com/74737261/182890370-584d5930-85ee-478a-bcf7-77d0846fc06a.png)
![image](https://user-images.githubusercontent.com/74737261/182890391-5cb909b1-2109-4852-aadc-900061b9f98b.png)
![image](https://user-images.githubusercontent.com/74737261/182890413-b1c2e546-bbcf-46b3-83e9-a5706b333554.png)
![image](https://user-images.githubusercontent.com/74737261/182890433-4d434a03-bac1-4221-b7e3-3b92f9a70551.png)

**Comment by amyegan** [comment]
Thank you for sharing all of this detail @bryanyanay! I am not able to see the error you're seeing since I do not have an account. What I do see on the login page is an alert in the upper right corner saying "One or more of your connections are currently using Auth0 development keys and should not be used in production" with a link to [learn more](https://auth0.com/docs/authenticate/identity-providers/social-identity-providers/devkeys). 

I tried to sign up for an account and got a [400 error](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) in the browser console from Auth0.

![image](https://user-images.githubusercontent.com/8247105/182950783-9c0f44db-47eb-447d-9476-4a212e9fa4b4.png)

All of this makes me think there is something wrong with one of your environment variables in your Vercel Project Settings. Can you double check that everything is set correctly? If you can share any other error details you're seeing in your browser's developer tools, that would also be helpful.

**Reply by bryanyanay** [reply]
Thanks for replying so quick! Also thanks for telling me about the development key alert and the 400 error; they should both be fixed now, they were both problems with how I set up auth0. You should be able to get to the "500 Internal Server Error" page now. 

I doublechecked the environment variables I set on vercel; they all matched the variables I'm using when I run the app on localhost (which works fine), _except_:
- `AUTH0_BASE_URL` is set to `http://localhost:3000` in my local env variables, but set to `https://power-transfer-app.vercel.app/` on my vercel env variables, which I believe is correct?
- on vercel, I have an env variable `AUTH0_SCOPE` set to `openid profile read:shows`, while locally i don't have that env variable. I added this env variable since [this readme](https://github.com/auth0/nextjs-auth0/blob/main/examples/README.md#environment-variables-1) seems to say I should.

As for devtool errors; on the 500 page, there's this error in the devtools console:
![image](https://user-images.githubusercontent.com/74737261/183215707-9d5b102d-4a22-4818-bc08-86ab4e3a40dc.png)
And when I go to the "Network" tab in devtools, I believe this is the request that generates that error (is that the right way to phrase it?):
![image](https://user-images.githubusercontent.com/74737261/183215932-46318a74-77a5-42cb-addd-b35f71b30a5b.png)

Once again, thank you so much for the help!

**Comment by swarnava** [comment]
Hi,

Thank you for sharing the details. I can see your initial screenshot was related to history.json and your latest screenshot is about `/vehicles` endpoint. Do you get any different error in Function logs after setting these env variable?

**Reply by bryanyanay** [reply]
Hi, thanks for the help!

All three of my pages other than index.tsx face the same "500 internal server error". (Specifically, the transfer, history, and vehicles pages.) That's why my first screenshot was for /vehicles, and my second is for /history. As far as I can tell, the pages seem to be failing for the same reasons (i.e., with basically the same error in the function logs).

Do you have any guess as to what the problem may be? A friend of mine suggested that the problem might be related to CORS?
